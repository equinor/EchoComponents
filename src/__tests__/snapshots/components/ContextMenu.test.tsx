import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonWithPopover, ContextMenu, DataInformation, getIcon } from '../../..';
import { themeConst } from '../../../theme/themeConst';

it('renders ContextMenu without more info button correctly', () => {
    const setExpanded = jest.fn();
    const openTagInformation = jest.fn();

    const tree = renderer
        .create(
            <ContextMenu
                icon={getIcon('Signal')}
                expanded={true}
                setExpanded={setExpanded}
                legendColor={themeConst.asBuilt}
                tagNo={'361B1-MA001-Q02'}
                description={'Pipehandling Crane Running'}
                positionStyle={{ left: '20px', top: '20px' }}
                openTagInformation={openTagInformation}
                selected={false}
            ></ContextMenu>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders ContextMenu with more info button correctly', () => {
    const setExpanded = jest.fn();
    const openTagInformation = jest.fn();
    const fetchDataToShow = jest.fn();
    const fetchedDataToShow: DataInformation[] = [];

    const tree = renderer
        .create(
            <ContextMenu
                icon={getIcon('Signal')}
                expanded={true}
                setExpanded={setExpanded}
                legendColor={themeConst.asBuilt}
                tagNo={'361B1-MA001-Q02'}
                description={'Pipehandling Crane Running'}
                positionStyle={{ left: '20px', top: '20px' }}
                openTagInformation={openTagInformation}
                selected={false}
            >
                <ButtonWithPopover
                    fetchDataToShow={fetchDataToShow}
                    isLoading={false}
                    fetchedData={fetchedDataToShow}
                />
            </ContextMenu>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
