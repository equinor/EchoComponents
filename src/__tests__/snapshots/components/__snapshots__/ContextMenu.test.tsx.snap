// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ContextMenu with more info button correctly 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "left": "20px",
      "top": "20px",
    }
  }
>
  <div
    className="contextWrapperExpanded"
    onClick={[MockFunction]}
  >
    <div
      className="tagInfoWrapper"
    >
      <div
        className="icon"
        style={
          Object {
            "background": "#007079",
          }
        }
      >
        <svg
          aria-label="signal"
        />
      </div>
      <div
        className="tagText"
      >
        <h5
          className="sc-eCssSg caHYEQ tagHeader"
        >
          361B1-MA001-Q02
        </h5>
        <p
          className="sc-eCssSg bHjGEK tagDescription"
        >
          Pipehandling Crane Running
        </p>
      </div>
      <div
        className="tagInfoIcon"
      >
        <button
          className="sc-hKgILt lkgudK"
          disabled={false}
          tabIndex={0}
          type="button"
        >
          <span
            className="sc-dlfnbm iOlYhO"
          >
            <svg
              aria-labelledby="eds-info_circle-2"
              className="sc-iwyYcG dqcEJR"
              fill="#007079"
              height="24px"
              name={null}
              role="img"
              title="tag information"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="eds-info_circle-2"
              >
                tag information
              </title>
              <path
                className="sc-cxFLnm"
                clipRule="evenodd"
                d="M12.01 22C17.53 22 22 17.52 22 12S17.53 2 12.01 2C6.48 2 2 6.48 2 12s4.48 10 10.01 10zM13 9V7h-2v2h2zm0 8v-6h-2v6h2zm7-5c0-4.42-3.58-8-8-8s-8 3.58-8 8 3.58 8 8 8 8-3.58 8-8z"
                data-testid="eds-icon-path"
                fillRule="evenodd"
                height="24"
                size={null}
                transform={null}
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="wrapper"
  >
    <button
      className="sc-hKgILt lkgudK button"
      disabled={false}
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="sc-dlfnbm iOlYhO"
      >
        <svg
          aria-labelledby="eds-more_horizontal-3"
          className="sc-iwyYcG dqcEJR"
          fill="#007079"
          height="24px"
          name={null}
          role="img"
          title="more options"
          viewBox="0 0 24 24"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="eds-more_horizontal-3"
          >
            more options
          </title>
          <path
            className="sc-cxFLnm"
            clipRule="evenodd"
            d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-8 2c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
            data-testid="eds-icon-path"
            fillRule="evenodd"
            height="24"
            size={null}
            transform={null}
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`renders ContextMenu without more info button correctly 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "left": "20px",
      "top": "20px",
    }
  }
>
  <div
    className="contextWrapperExpanded"
    onClick={[MockFunction]}
  >
    <div
      className="tagInfoWrapper"
    >
      <div
        className="icon"
        style={
          Object {
            "background": "#007079",
          }
        }
      >
        <svg
          aria-label="signal"
        />
      </div>
      <div
        className="tagText"
      >
        <h5
          className="sc-eCssSg caHYEQ tagHeader"
        >
          361B1-MA001-Q02
        </h5>
        <p
          className="sc-eCssSg bHjGEK tagDescription"
        >
          Pipehandling Crane Running
        </p>
      </div>
      <div
        className="tagInfoIcon"
      >
        <button
          className="sc-hKgILt lkgudK"
          disabled={false}
          tabIndex={0}
          type="button"
        >
          <span
            className="sc-dlfnbm iOlYhO"
          >
            <svg
              aria-labelledby="eds-info_circle-1"
              className="sc-iwyYcG dqcEJR"
              fill="#007079"
              height="24px"
              name={null}
              role="img"
              title="tag information"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="eds-info_circle-1"
              >
                tag information
              </title>
              <path
                className="sc-cxFLnm"
                clipRule="evenodd"
                d="M12.01 22C17.53 22 22 17.52 22 12S17.53 2 12.01 2C6.48 2 2 6.48 2 12s4.48 10 10.01 10zM13 9V7h-2v2h2zm0 8v-6h-2v6h2zm7-5c0-4.42-3.58-8-8-8s-8 3.58-8 8 3.58 8 8 8 8-3.58 8-8z"
                data-testid="eds-icon-path"
                fillRule="evenodd"
                height="24"
                size={null}
                transform={null}
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
